mautic/mautic
# Feature ux bulk contact preferences update

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y |
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Required: )
#### Description:
Implementation of batch updates from contact's preferences. Huge refactoring included. Unit tests included.

#### Steps to test this PR:
1. Make sure that bulk update of segments, categories and channels in contacts is working properly
2. Try that contact preference update is working as well (no visible changes)
